flash/tabular/regression/data.py

```diff
@@ -11,8 +11,165 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from flash.tabular.data import TabularData
+from typing import Any, Callable, Dict, List, Optional, Union
+
+from flash.core.data.callback import BaseDataFetcher
+from flash.core.data.process import Deserializer
+from flash.core.utilities.imports import _PANDAS_AVAILABLE
+
+if _PANDAS_AVAILABLE:
+    from pandas.core.frame import DataFrame
+else:
+    DataFrame = object
+
+from pytorch_forecasting import TimeSeriesDataSet
+
+from flash.tabular.data import TabularData, TabularDataFrameDataSource, TabularPreprocess
+
+
+class TabularRegressionDataFrameDataSource(TabularDataFrameDataSource):

```

Comment:
Maybe this would be better just extending DataSource as it doesn't use any of the functionality from the TabularDataFrameDataSource